ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/sync/SynchronizerConfiguration.java

```diff
@@ -216,24 +204,36 @@ public int getMaxTrailingPeers() {
   public static class Builder {
     private SyncMode syncMode = SyncMode.FULL;
     private Range<Long> blockPropagationRange = Range.closed(-10L, 30L);
-    private long downloaderChangeTargetThresholdByHeight = 20L;
-    private UInt256 downloaderChangeTargetThresholdByTd = UInt256.of(1_000_000_000L);
-    private int downloaderHeaderRequestSize = 200;
-    private int downloaderCheckpointTimeoutsPermitted = 5;
-    private int downloaderChainSegmentTimeoutsPermitted = 5;
-    private int downloaderChainSegmentSize = 200;
-    private int downloaderParallelism = 4;
-    private int transactionsParallelism = 2;
-    private int computationParallelism = Runtime.getRuntime().availableProcessors();
-    private int fastSyncPivotDistance = DEFAULT_PIVOT_DISTANCE_FROM_HEAD;
-    private float fastSyncFullValidationRate = DEFAULT_FULL_VALIDATION_RATE;
-    private int fastSyncMinimumPeerCount = DEFAULT_FAST_SYNC_MINIMUM_PEERS;
-    private int worldStateHashCountPerRequest = DEFAULT_WORLD_STATE_HASH_COUNT_PER_REQUEST;
-    private int worldStateRequestParallelism = DEFAULT_WORLD_STATE_REQUEST_PARALLELISM;
+    private long downloaderChangeTargetThresholdByHeight =
+        SynchronizerPerformanceOptions.get().downloaderChangeTargetThresholdByHeight;
```

Comment:
Using the singleton here probably is bad since `SynchronizerConfig.Builder` is used when creating multiple instances in the same VM.  We probably want it to be completely isolated from the command line parsing.  Probably going to create a bunch more boiler plate unfortunately. I think having a clear separation between command line and runtime code is probably worth it.

Reply:
This may be the correct time to look at DI again.

Reply:
Maybe, but if you can do it with DI you can do it without - it's just a question of how many places you have to pass the value through.  Showing the two different approaches in this particular situation is probably a good way to evaluate DI...

Reply:
+1 for the DI evaluation